src/common/k8s-api/kube-api.ts

```diff
@@ -156,10 +156,7 @@ export class KubeApi<T extends KubeObject> {
       isNamespaced = options.objectConstructor?.namespaced
     } = options || {};
 
-    if (!options.apiBase) {
-      options.apiBase = objectConstructor.apiBase;
-    }
-    const { apiBase, apiPrefix, apiGroup, apiVersion, resource } = parseKubeApi(options.apiBase);
+    const { apiBase, apiPrefix, apiGroup, apiVersion, resource } = parseKubeApi(options.apiBase || objectConstructor.apiBase);
```

Comment:
We don't anymore mutate `options`.. was that intentional / does it cause any side-effects?

Reply:
You added it in https://github.com/lensapp/lens/pull/1125 so I don't know if it is intentional. 

Technically this change could be outside observable because we pass in an object reference as options however I don't think we should consider this a breaking change because reusing an options object after using it seems like a definitely unsupported use case. Other than that this shouldn't have any side effects (unless someone passes in a proxy object and want to react to this specific assignment for some reason).